### PR TITLE
Feature/vendor

### PR DESCRIPTION
**Pull request by arijit1111**
Vendor Registration with Policy Authentication, Authorization,
Vendor Account Update
Migrations for User, Brand

### CODE REVIEW

File: Modules/Brand/Entities/Brand.php [hunk 1]
```diff
@@ -3,9 +3,11 @@
 namespace Modules\Brand\Entities;
 
 use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Str;
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Entities/Brand.php]: Remove this line

File: Modules/Brand/Entities/Brand.php [hunk 2]
```diff
@@ -3,9 +3,11 @@
...
 
 class Brand extends Model {
 
     protected $guarded = [];
 
+
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Entities/Brand.php]: remove this line

File: Modules/Brand/Http/Services/BrandService.php [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+use Illuminate\Support\Str;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rules\Password;
+use Modules\Brand\Entities\Catalog;
+use Modules\Country\Entities\Country;
+use Modules\User\Entities\User;
+use Modules\Brand\Entities\Brand;
+use File;
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php]: I think this should be Illuminate\Support\Facades\File;

File: Modules/Brand/Http/Services/BrandService.php [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+        $brand->product_shipped = $productShippedCountry->name;
+
+        return [
+            'res' => true,
+            'msg' => '',
+            'data' => $brand
+        ];
+    }
+
+    /**
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php:215]: description is missing

File: Modules/Brand/Http/Services/BrandService.php [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+                $brand->save();
+            }
+            if ($request->file('upload_contact_list')) {
+                $fileName = Str::random(10) . '_cstmrs.' . $request->file('upload_contact_list')->extension();
+                $request->file('upload_contact_list')->move($brandAbsPath, $fileName);
+                $brand->upload_contact_list = $brandRelPath . $fileName;
+                $brand->save();
+            }
+            $response = ['res' => true, 'msg' => "", 'data' => $data];
+        } catch (Exception $e) {
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php:309]: should be \Exception not Exception

File: Modules/Brand/Http/Services/BrandService.php [hunk 6]
```diff
@@ -0,0 +1,509 @@
...
+
+        if ($replaceable && $previousFile !== null) {
+            $unlinkUrl = public_path() . $previousFile;
+            if (file_exists($unlinkUrl)) {
+                unlink($unlinkUrl);
+            }
+        }
+
+        $image_64 = $image; //your base64 encoded data
+        $extension = explode('/', explode(':', substr($image_64, 0, strpos($image_64, ';')))[1])[1];   // .jpg .png .pdf
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php]: unused variable, remove

File: Modules/Brand/Http/Services/BrandService.php [hunk 7]
```diff
@@ -0,0 +1,509 @@
...
+    }
+
+    /**
+     * Save image from base64 string.
+     *
+     * @param int $brand
+     * @param $image
+     * @param $previousFile
+     * @param $replaceable
+     * @return Stringable
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php]: Stringable is not used

File: Modules/Brand/Http/Services/BrandService.php [hunk 8]
```diff
@@ -0,0 +1,509 @@
...
+        $extension = explode('/', explode(':', substr($image_64, 0, strpos($image_64, ';')))[1])[1];   // .jpg .png .pdf
+        $replace = substr($image_64, 0, strpos($image_64, ',') + 1);
+        $image_64 = str_replace($replace, '', $image_64);
+        $image_64 = str_replace(' ', '+', $image_64);
+        $imageName = Str::random(10) . '.' . 'png';
+        File::put($brandAbsPath . $imageName, base64_decode($image_64));
+        return $brandRelPath . $imageName;
+    }
+
+    public function updateAccount($request): array
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php]: Missing comment

File: Modules/Brand/Http/Services/BrandService.php [hunk 9]
```diff
@@ -0,0 +1,509 @@
...
+    {
+        $validator = Validator::make($request->all(), [
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php]: Validation need to be done in request
Authorization is missing

File: Modules/Brand/Http/Services/BrandService.php [hunk 10]
```diff
@@ -0,0 +1,509 @@
...
+            } else {
+                $response = ['res' => false, 'msg' => "Please try again!", 'data' => ''];
+            }
+        }
+
+        return $response;
+    }
+
+    /**
+     * @param $request
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php]: Missing comment

File: Modules/Brand/Http/Services/BrandService.php [hunk 11]
```diff
@@ -0,0 +1,509 @@
...
+     * @return array
+     */
+    public function updateShop($request): array
+    {
+        $userId = $request->user_id;
+        $brand = Brand::where('user_id', $request->user_id)->first();
+        $brandId = $brand->id;
+        $request->brand_slug = Str::slug($request->brand_name, '-');
+        $validator = Validator::make($request->all(), [
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php]: Need to be done in request

File: Modules/Brand/Routes/api.php [hunk 12]
```diff
@@ -13,31 +14,23 @@
...
-Route::prefix('brand')->group(function () {
-    Route::get('/all/{retailer}', 'BrandController@all');
-});
-Route::prefix('brand')->group(function () {
-    Route::get('/customers', 'BrandController@customers');
-});
-Route::prefix('brand')->group(function () {
-    Route::post('/add-customer', 'BrandController@addCustomer');
-});
+Route::prefix('brands')
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Routes/api.php:17]: Shouldn't this be under auth middleware?

**Review by SonaKhachatryan1995 — CHANGES_REQUESTED** [review]

File: Modules/Brand/Entities/Brand.php (hunk 1 above)
Comment (arijit1111) [reply Modules/Brand/Entities/Brand.php]: done

File: Modules/Brand/Entities/Brand.php (hunk 2 above)
Comment (arijit1111) [reply Modules/Brand/Entities/Brand.php]: done

File: Modules/Brand/Http/Services/BrandService.php (hunk 4 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php:215]: done

File: Modules/Brand/Http/Services/BrandService.php (hunk 3 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php]: done

File: Modules/Brand/Http/Services/BrandService.php (hunk 5 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php:309]: done

File: Modules/Brand/Http/Services/BrandService.php (hunk 6 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php]: done

File: Modules/Brand/Http/Services/BrandService.php (hunk 7 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php]: done

File: Modules/Brand/Http/Services/BrandService.php (hunk 8 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php]: done

File: Modules/Brand/Http/Services/BrandService.php (hunk 10 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php]: done

File: Modules/Brand/Routes/api.php (hunk 12 above)
Comment (arijit1111) [reply Modules/Brand/Routes/api.php:17]: during registration of brand due to many steps some routes are accessed without login

File: Modules/Brand/Http/Services/BrandService.php (hunk 9 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php]: done

File: Modules/Brand/Http/Services/BrandService.php (hunk 11 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php]: done

File: Modules/Brand/Http/Controllers/BrandController.php [hunk 13]
```diff
@@ -5,20 +5,13 @@
...
-use Modules\Cart\Entities\Cart;
 use Modules\Brand\Http\Requests\StoreBrandRequest;
 use Modules\Brand\Http\Requests\UpdateBrandRequest;
+use Modules\Brand\Http\Requests\UpdateBrandAccountRequest;
+use Modules\Brand\Http\Requests\UpdateBrandShopRequest;
 use Modules\Brand\Http\Services\BrandService;
-use File;
-use DB;
-use Exception;
+
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Controllers/BrandController.php:15]: Add use Modules\Brand\Entities\Brand;
we are using Brand in liveShop and other d=functions, but not included it.

File: Modules/Brand/Http/Services/BrandService.php [hunk 14]
```diff
@@ -4,22 +4,22 @@
 
 use Illuminate\Support\Str;
 use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\File;
 use Illuminate\Support\Facades\Hash;
 use Illuminate\Support\Facades\Validator;
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php]: do not need this anymore, delete

File: Modules/Brand/Http/Services/BrandService.php [hunk 15]
```diff
@@ -4,22 +4,22 @@
...
 use Illuminate\Validation\Rules\Password;
```
Comment (SonaKhachatryan1995) [inline Modules/Brand/Http/Services/BrandService.php]: do not need this anymore, delete

**Review by SonaKhachatryan1995 — CHANGES_REQUESTED** [review]

File: Modules/Brand/Http/Controllers/BrandController.php (hunk 13 above)
Comment (arijit1111) [reply Modules/Brand/Http/Controllers/BrandController.php:15]: done

File: Modules/Brand/Http/Services/BrandService.php (hunk 14 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php]: done

Comment (SonaKhachatryan1995) [reply Modules/Brand/Http/Services/BrandService.php]: comment was related to use Illuminate\Support\Facades\Validator; not use Illuminate\Support\Facades\File;
bring use Illuminate\Support\Facades\File; back and delete
use Illuminate\Support\Facades\Validator;

File: Modules/Brand/Http/Services/BrandService.php (hunk 15 above)
Comment (SonaKhachatryan1995) [reply Modules/Brand/Http/Services/BrandService.php]: comment was related to use Illuminate\Validation\Rules\Password; not use Illuminate\Support\Facades\File;
bring use Illuminate\Support\Facades\File; back and delete
use Illuminate\Validation\Rules\Password;

File: Modules/Brand/Http/Services/BrandService.php (hunk 14 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php]: Fixed

File: Modules/Brand/Http/Services/BrandService.php (hunk 15 above)
Comment (arijit1111) [reply Modules/Brand/Http/Services/BrandService.php]: Fixed

**Review by SonaKhachatryan1995 — APPROVED** [review]

**Pull request merged**